### DISCUSSION TITLE
Detecting a box in bottom-right corner of image and cropping image to it

### DISCUSSION THREAD

**Discussion by forsberg72** [ImageMagick/ImageMagick#5284] [Help] [answered]
I have hundreds of images like this: https://ibb.co/pLthkX6

All the images have a rectangle around the whole drawing, and all have also that rectangle in bottom right corner. That rectangle contains smaller rectangles which contain text. I marked the rectangle with red arrow. How could I possibly crop images so that I only get that rectangle containing the important texts which I could possibly extract with Tesseract or similar?

Images vary in size and they can be landscape or portrait. I'm familiar with scripting/programming (e.g. Perl, Bash, etc.) and basic usage of IM tools, but I don't know advanced use of IM.

**Comment by snibgo** [comment]
For problems like this, the hard part is defining the rules that tell us where the label is. Once we know the rules, making an ImageMagick command is relatively simple.

I hope your images have more resolution than that example. The letters in the box are only about 6 pixels high, which will be a problem for OCR.

The file is PNG but I suspect the image has been through JPEG compression, which makes the task more difficult.

Can we assume the label is always bottom-right? And that it is always less than half the height of the overall drawing, and less than half of the width? If so, this reduces the search space ("-crop 50x50%+0+0 +repage").

For this drawing, a method might:

1. Find the right-most and bottom-most long lines.

2. The outline of the label, and internal lines within the label, are darker than any other lines attached to the lines from (1). So we can find those darkest lines.

For (1), we can (a) make short lines white (including the spurious line on the right), and trim white space.

For (2), we can make light gray pixels white ("-threshold"). Then find T-junctions ("-morphology"). The left-most inverted T ...
```
   |
   |
---+---
```
... defines the left edge of the label. The top-most rotated T...
```
   |
   |
---+
   |
   |
```
... defines the top of the label.

**Reply by forsberg72** [reply]
Thanks for the answer! Images are originally very high resolution TIFFs, so there are no compression artifacts and text seems to be readable for tesseract.

**Comment by snibgo** [comment, marked as answer]
For the image you supply, the following (Windows BAT syntax) does the job:
```
set KNL=^
1,1,1,1,^
1,-,0,0,^
1,0,1,1,^
1,0,1,1

magick ^
  kuvatus.png ^
  -gravity SouthEast ^
  -crop 50x50%%+0+0 ^
  -format "%%O\n" +write info: +format ^
  +repage ^
  -threshold 25%% ^
  -morphology HMT "%KNL%" ^
  -define "identify:locate=maximum" ^
  info:
```
The text output is:
```
+550+346
Channel maximum locations:
  Red: 69904 (1.06667) 242,251
  Green: 69904 (1.06667) 242,251
  Blue: 69904 (1.06667) 242,251
```
We cropped the south-east corner, and that starts at (550,346). Then we thresholded, and searched for a corner piece (the "kernel"). That was found at (242,251) in the crop, so we need to add the coordinates to get the location (792,597) of the top-left of the label in the input image. So now we can crop the label from the original image:
```
magick ^
  kuvatus.png ^
  -crop +792+597 +repage ^
  c.png
```
c.png is:
![c](https://user-images.githubusercontent.com/33812368/176814474-c6a12240-1964-4eac-a498-47451becccf7.png)

We could then use "-morphology" to remove long horizontal and vertical lines.

I expect that simple kernel wouldn't work on your large TIFF originals. Other changes may also be needed.
